code/src/pages/product-details/components/product-details-container/ProductDetailsContainer.tsx

```diff
@@ -0,0 +1,154 @@
+import { useIntl } from "react-intl";
+
+import AppBadge from "@/components/app-badge/AppBadge";
+import AppBox from "@/components/app-box/AppBox";
+import AppButton from "@/components/app-button/AppButton";
+import AppTypography from "@/components/app-typography/AppTypography";
+
+import { deliveryMethods as deliveryMethodsData } from "@/constants/deliveryMethods";
+import { useLocaleContext } from "@/context/i18n/I18nProvider";
+import useErrorPageRedirect from "@/hooks/use-error-page-redirect/useErrorPageRedirect";
+import { productNotFoundRedirectConfig } from "@/pages/product-details/ProductsDetailsPage.constants";
+import { useGetUserProductByIdQuery } from "@/store/api/productsApi";
+import formatPrice from "@/utils/format-price/formatPrice";
+import getCategoryFromTags from "@/utils/get-category-from-tags/getCategoryFromTags";
+import isErrorWithStatus from "@/utils/is-error-with-status/isErrorWithStatus";
+
+import "@/pages/product-details/components/product-details-container/ProductDetailsContainer.scss";
+
+type ProductDetailsContainerProps = {
+  productId: string;
+};
+
+const ProductDetailsContainer = ({
+  productId
+}: ProductDetailsContainerProps) => {
+  const { renderRedirectComponent } = useErrorPageRedirect();
+  const { locale } = useLocaleContext();
+  const { formatMessage } = useIntl();
+  const {
+    data: product,
+    isLoading,
+    error
+  } = useGetUserProductByIdQuery({
+    productId,
+    lang: locale
+  });
+
+  // @TODO: replace with actual loading fallback
+  if (isLoading) {
+    return <div>Loading...</div>;
+  }
+
+  const isNotFoundOnServer = isErrorWithStatus(error) && error.status === 404;
+
+  if (isNotFoundOnServer) {
+    return renderRedirectComponent(productNotFoundRedirectConfig);
+  }
+
+  // @TODO: add error handling
+  if (!product || error) {
+    return <div>Error...</div>;
+  }
+
+  const categoryTag = getCategoryFromTags(product.tags);
+
+  const categoryBadge = categoryTag && (
+    <AppBadge
+      badgeContent={
+        <AppTypography
+          variant="caption-small"
+          translationKey={`productsAll.${categoryTag}`}
+        />
+      }
+    />
+  );
+
+  const productDescription = product.description
+    .split("/")
+    .map((paragraph) => (
+      <AppTypography key={paragraph}>{paragraph}</AppTypography>
+    ));
```

Comment:
What do you think about the creating function for that?

Reply:
Thank you for the question! I don't think that we need to create some util for it. I don't see any use case in another component for now. If I am wrong, please correct me